test/integration/resourcemanager/podzoneaffinity/podzoneaffinity_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podzoneaffinity_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("PodSchedulerName tests", func() {
+	var (
+		pod *corev1.Pod
+	)
+
+	BeforeEach(func() {
+		pod = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-",
+				Namespace:    testNamespace.Name,
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  "foo-container",
+						Image: "foo",
+					},
+				},
+			},
+		}
+		DeferCleanup(func() {
+			Expect(testClient.Delete(ctx, pod)).To(Succeed())
+		})
+	})
+
+	Context("when namespace has zone enforcement label", func() {
+		BeforeEach(func() {
+			testNamespace.Labels = map[string]string{
+				"control-plane.shoot.gardener.cloud/enforce-zone": "",
+			}
+			Expect(testClient.Update(ctx, testNamespace)).To(Succeed())
+			DeferCleanup(func() {
+				testNamespace.Labels = nil
+				Expect(testClient.Update(ctx, testNamespace)).To(Succeed())
+			})
+		})
+
+		It("should add podAffinity", func() {
+			Expect(testClient.Create(ctx, pod)).To(Succeed())
+
+			Consistently(func() ([]corev1.PodAffinityTerm, error) {
```

Comment:
Why would we have to use async assertions in such webhook tests? I see that you probably copied this from other existing tests, but shouldn't the `pod` already have been modified right during creation (otherwise, the above `Create` call failed)?
If you just copied this and there is no reason for it, please remove it (maybe also cleanup the other test suites as you are on it) 👀